core-bundle/src/Resources/contao/dca/tl_opt_in.php

```diff
@@ -0,0 +1,206 @@
+		),
+		'operations' => array
+		(
+			'resend' => array
+			(
+				'label'               => &$GLOBALS['TL_LANG']['tl_opt_in']['resend'],
+				'href'                => 'key=resend',
+				'icon'                => 'resend.svg',
+				'button_callback'     => array('tl_opt_in', 'resendButton')
+			),
+			'show' => array
+			(
+				'label'               => &$GLOBALS['TL_LANG']['tl_opt_in']['show'],
+				'href'                => 'act=show',
+				'icon'                => 'show.svg'
+			)
+		)
+	),
+
+	// Fields
+	'fields' => array
+	(
+		'id' => array
+		(
+			'sql'                     => "int(10) unsigned NOT NULL auto_increment"
+		),
+		'tstamp' => array
+		(
+			'sql'                     => "int(10) unsigned NOT NULL default '0'"
+		),
+		'token' => array
+		(
+			'label'                   => &$GLOBALS['TL_LANG']['tl_opt_in']['token'],
+			'search'                  => true,
+			'sql'                     => "varchar(24) NOT NULL default ''"
+		),
+		'createdOn' => array
+		(
+			'label'                   => &$GLOBALS['TL_LANG']['MSC']['createdOn'],
+			'filter'                  => true,
+			'sorting'                 => true,
+			'flag'                    => 6,
+			'eval'                    => array('rgxp'=>'datim'),
+			'sql'                     => "int(10) unsigned NOT NULL default '0'"
+		),
+		'confirmedOn' => array
+		(
+			'label'                   => &$GLOBALS['TL_LANG']['tl_opt_in']['confirmedOn'],
+			'filter'                  => true,
+			'sorting'                 => true,
+			'flag'                    => 6,
+			'eval'                    => array('rgxp'=>'datim'),
+			'sql'                     => "int(10) unsigned NOT NULL default '0'"
+		),
+		'removeOn' => array
+		(
+			'label'                   => &$GLOBALS['TL_LANG']['tl_opt_in']['removeOn'],
+			'filter'                  => true,
+			'sorting'                 => true,
+			'flag'                    => 6,
+			'eval'                    => array('rgxp'=>'datim'),
+			'sql'                     => "int(10) unsigned NOT NULL default '0'"
+		),
+		'email' => array
+		(
+			'label'                   => &$GLOBALS['TL_LANG']['MSC']['emailAddress'],
+			'search'                  => true,
+			'sorting'                 => true,
+			'eval'                    => array('rgxp'=>'email'),
+			'sql'                     => "varchar(255) NOT NULL default ''"
+		),
+		'emailSubject' => array
+		(
+			'label'                   => &$GLOBALS['TL_LANG']['tl_opt_in']['emailSubject'],
+			'search'                  => true,
+			'sql'                     => "varchar(255) NOT NULL default ''"
+		),
+		'emailText' => array
+		(
+			'label'                   => &$GLOBALS['TL_LANG']['tl_opt_in']['emailText'],
+			'search'                  => true,
+			'sql'                     => "text NULL"
+		)
+	)
+);
+
+/**
+ * Provide miscellaneous methods that are used by the data configuration array.
+ *
+ * @author Leo Feyer <https://github.com/leofeyer>
+ */
+class tl_opt_in extends Backend
```

Comment:
Please, we don't do this anymore. We can keep the old code where it is but for new stuff, let's use proper listeners. They can even register on callbacks in `services.yml` now.

Reply:
Consistent beats nice. 🙈

Reply:
No, really not. We have to lead by example. We keep saying that things are still the old way because of BC. Here there is no BC and there's no reason to put a class into a config file whatsoever.